app/src/main/java/com/rockthevote/grommet/ui/registration/personal/AdditionalInfoExt.kt

```diff
@@ -0,0 +1,71 @@
+package com.rockthevote.grommet.ui.registration.personal
+
+import com.rockthevote.grommet.data.db.model.Party
+import com.rockthevote.grommet.data.db.model.PhoneType
+import com.rockthevote.grommet.data.db.model.PreferredLanguage
+import com.rockthevote.grommet.data.db.model.Race
+import com.rockthevote.grommet.databinding.FragmentAdditionalInfoBinding
+import com.rockthevote.grommet.ui.registration.name.toEditable
+
+fun FragmentAdditionalInfoBinding.toAdditionalInfoData(): AdditionalInfoData {
+    val party = spinnerParty.spinnerText.toString().let {
+        Party.fromString(it)
+    }
+
+    val emailAddress = emailEditText.text.toString()
+    val phoneNumber = phone.text.toString()
+    val phoneType = spinnerPhoneType.spinnerText.toString().let {
+        PhoneType.fromString(it)
+    }
+
+    val hasChangedPoliticalParty = politicalPartyChangeTextbox.isChecked
+    val knowsPennDotNumber = !doesNotHavePennDotCheckbox.isChecked
+    val knowsSsnLastFour = !ssnLastFourCheckbox.isChecked
+    val hasOptedIntoNewsUpdates = emailOptIn.isChecked
+    val hasOptedIntoNewsCallAndText = checkboxCanReceiveText.isChecked
+    val hasOptedForVolunteerText = checkboxPartnerVolunteerOptIn.isChecked
+    val preferredLanguage = spinnerPreferredLanguage.spinnerText?.let {
+        PreferredLanguage.fromString(it)
+    }
+    val race = spinnerRace.spinnerText?.let { Race.fromString(it) }
+
+    val otherPoliticalParty = if (party == Party.OTHER_PARTY) otherPartyEditText.text?.toString() else null
```

Comment:
good job spotting all these "one-offs".

Reply:
I hope I got all of them, I tested this a few times through as thoroughly as I knew how.

Reply:
i've been keeping track of some odd ones along the way and will double check at the end